pyanaconda/modules/payloads/source/cdrom/initialization.py

```diff
@@ -41,14 +43,58 @@ def _do_mount(self):
         log.debug("Trying to detect CD-ROM automatically")
 
         device_tree = STORAGE.get_proxy(DEVICE_TREE)
+
+        device_candidates = device_tree.FindOpticalMedia()
+        stage2_device = self._probe_stage2_for_cdrom(device_tree)
+
+        if stage2_device:
+            device_candidates = [stage2_device] + device_candidates
```

Comment:
I think that this should be in another function that will return the final device candidates.